lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1782,30 +1910,27 @@ static bool isExpressionEmittedInline(Operation *op) {
   return op->getResult(0).hasOneUse();
 }
 
-// Print out the array subscripts after a wire/port declaration.
-static void printArraySubscripts(Type type, raw_ostream &os) {
-  if (auto inout = type.dyn_cast<InOutType>())
-    return printArraySubscripts(inout.getElementType(), os);
-
-  if (auto array = type.dyn_cast<UnpackedArrayType>()) {
-    printArraySubscripts(array.getElementType(), os);
-    os << '[' << (array.getSize() - 1) << ":0]";
-  }
-}
-
 void ModuleEmitter::collectNamesEmitDecls(Block &block) {
   // In the first pass, we fill in the symbol table, calculate the max width
   // of the declaration words and the max type width.
   size_t maxDeclNameWidth = 0, maxTypeWidth = 0;
 
   // Return the word (e.g. "wire") in Verilog to declare the specified thing.
-  auto getVerilogDeclWord = [](Operation *op) -> StringRef {
+  // Structs are computed, so we can't use StringRef
+  auto getVerilogDeclWord = [](Operation *op) -> std::string {
     if (isa<RegOp>(op))
       return "reg";
 
     // Interfaces instances use the name of the declared interface.
     if (auto interface = dyn_cast<InterfaceInstanceOp>(op))
-      return interface.getInterfaceType().getInterface().getValue();
+      return interface.getInterfaceType().getInterface().getValue().str();
+
+    if (auto wire = dyn_cast<WireOp>(op)) {
+      std::string str;
+      llvm::raw_string_ostream stringStream(str);
+      baseType(wire.getElementType(), "wire", op->getLoc(), stringStream);
+      return str;
```

Comment:
Please find some other way to structure this code, keeping `getVerilogDeclWord` true to its name.  I think this logic should be outside of it, or we should rename getVerilogDeclWord.  Also, please don't use streams to create strings that get dumped into streams.  :-)